devicetypes/fibargroup/fibaro-door-window-sensor-2.src/fibaro-door-window-sensor-2.groovy

```diff
@@ -129,12 +131,26 @@ def updated() {
 	}
 	logging("${device.displayName} - Executing updated()","debug")
 
-	if ( settings.temperatureHigh as Integer == 0 && settings.temperatureLow as Integer == 0 ) { 
-		sendEvent(name: "temperatureAlarm", value: null, displayed: false) 
-	} else if ( settings.temperatureHigh != null || settings.temperatureHigh != null ) {
-		sendEvent(name: "temperatureAlarm", value: "clear", displayed: false) 
+	def tempAlarmMap = ["clear": "cleared", "overheat": "heat", "underheat": "freeze"]
+	// Convert old device specific temperatureAlarm event to standard Temperature Alarm capability event
+	def temperatureAlarmCurrentValue = device.currentValue("temperatureAlarm")
+	if (tempAlarmMap.containsKey(temperatureAlarmCurrentValue)) {
+		sendEvent(name: "temperatureAlarm", value: tempAlarmMap[temperatureAlarmCurrentValue], displayed: true)
 	}
-	
+
+	def currentTemperature = device.currentValue("temperature")
+	def alarmCleared = device.currentValue("temperatureAlarm") == "cleared"
+	def alarmFreeze = device.currentValue("temperatureAlarm") == "freeze"
+	def alarmHeat = device.currentValue("temperatureAlarm") == "heat"
+	def temperatureHigh = (settings.temperatureHigh ? new BigDecimal(settings.temperatureHigh) : null) * 0.1
+	def temperatureLow = (settings.temperatureLow ? new BigDecimal(settings.temperatureLow) : null) * 0.1

```

Comment:
won't you get an error here multiplying null by 0.1?

I think you should be using a default value here instead.

Reply:
We need to multiply these values due to the correct comparison _currentTemperature < temperatureHigh  /  currentTemperature > temperatureLow_. This is needed because values ​​multiplied by 10 are sent to the device (because settings are defined in this way and the device accepts and reports this way as well)